bin/named-person-data-export.js

```diff
@@ -0,0 +1,204 @@
+
+    // Fetch activity logs based on the profile_ids
+    const queryActivityLog = await getActivityLogs(profileIds);
+    console.log('Activity Logs from DB Taskflow:', queryActivityLog.length);
+
+    // Extract case_ids from activity logs
+    const caseIds = [...new Set(queryActivityLog.map(log => log.case_id))]; // Remove duplicates
+
+    // Fetch cases based on the extracted case_ids
+    const queryCases = await getCases(caseIds);
+    console.log('Cases from DB Taskflow:', queryCases.length);
+
+    // Convert cases array to a map for quick lookup
+    const casesMap = queryCases.reduce((acc, caseData) => {
+      acc[caseData.case_id] = caseData;
+      return acc;
+    }, {});
+
+    // Prepare the final result
+    const result = [];
+
+    // Keep track of cases already added per profile
+    const profileCaseMap = new Map();
+
+    profiles.forEach(profile => {
+      // Ensure each profile only gets a case once, regardless of organization
+      if (!profileCaseMap.has(profile.profile_id)) {
+        profileCaseMap.set(profile.profile_id, new Set());
+      }
+
+      queryActivityLog.forEach(log => {
+        if (log.changed_by === profile.profile_id) {
+          const caseData = casesMap[log.case_id];
+
+          if (caseData && !profileCaseMap.get(profile.profile_id).has(log.case_id)) {
+            result.push({
+              profile_id: profile.profile_id,
+              title: profile.title,
+              first_name: profile.first_name,
+              last_name: profile.last_name,
+              email: profile.email,
+              telephone: profile.telephone,
+              admin: profile.admin,
+              roles: profile.roles,
+              status: profile.status,
+              cases: `Case ID: ${caseData.case_id}, Status: ${caseData.status}, Model: ${caseData.model}, Model Status: ${caseData.model_status}`
+            });
+
+            profileCaseMap.get(profile.profile_id).add(log.case_id); // Prevent duplicate cases for this profile
+          }
+        }
+      });
+    });
+
+    // Check the result
+    console.log('Merged Result:', result);
+
+    // Write the result to CSV
+    const csvStream = fastCsv.format({
+      headers: [
+        'profile_id', 'title', 'first_name', 'last_name', 'email', 'telephone', 'admin', 'roles', 'status', 'cases'
+      ]
+    });
+
+    const writableStream = fs.createWriteStream('merged_profiles_cases.csv');
+
+    writableStream.on('finish', () => {
+      console.log('\nCSV file merged_profiles_cases.csv created with', result.length, 'rows');
+    });
+
+    csvStream.pipe(writableStream);
+    result.forEach(row => csvStream.write(row)); // Write each row
+    csvStream.end();
+  } catch (error) {
+    console.error('Error merging data:', error);
+  } finally {
+    console.log('Cleaning up...');
+    await knexASL.destroy();
+    await knexTaskflow.destroy();
+  }
+}
+
+// Run the script
+mergeAndSaveCSV();
```

Comment:
I think this needs to be awaited to make sure it runs to completion, but I might be wrong. Node should support top level await now.

```suggestion
await mergeAndSaveCSV();
```

Reply:
I am not sure about this, the function is asynchronous so it should follow the asynchronous approach by default!

Reply:
it causing issues with CommonJs. Need more time to look at!

Reply:
Perhaps we can move the whole script to a route for download if we need this data in future?